src/main/java/org/candlepin/model/DeletedConsumerCurator.java

```diff
@@ -94,9 +104,17 @@ public int countByConsumer(Consumer c) {
     }
 
     public int countByConsumerUuid(String uuid) {
-        return ((Long) currentSession().createCriteria(DeletedConsumer.class)
-            .add(Restrictions.eq("consumerUuid", uuid))
-            .setProjection(Projections.rowCount()).uniqueResult()).intValue();
+        if (uuid == null || uuid.isBlank()) {
+            return 0;
+        }
+
+        String query = "SELECT COUNT(dc) FROM DeletedConsumer dc WHERE dc.consumerUuid = :uuid";
+
+        return this.getEntityManager()
+            .createQuery(query, Number.class)
```

Comment:
Can you explain the usage of `Number.class` here instead of a more specific class?

Reply:
Oops, this should be `Long.class` because the value returned from the query is a Long. Both work, but since we know the value is a Long, it is better to use the subclass.